packages/stratimikos/adapters/belos/src/Thyra_BelosLinearOpWithSolveFactory_def.hpp

```diff
@@ -367,87 +350,12 @@ BelosLinearOpWithSolveFactory<Scalar>::generateAndGetValidParameters()
 {
   using Teuchos::as;
   using Teuchos::tuple;
-  using Teuchos::setStringToIntegralParameter;
-Teuchos::ValidatorXMLConverterDB::addConverter(
-  Teuchos::DummyObjectGetter<
-    Teuchos::StringToIntegralParameterEntryValidator<EBelosSolverType> 
-  >::getDummyObject(),
-  Teuchos::DummyObjectGetter<Teuchos::StringToIntegralValidatorXMLConverter<
-    EBelosSolverType> >::getDummyObject());
 
   typedef MultiVectorBase<Scalar> MV_t;
   typedef LinearOpBase<Scalar> LO_t;
   static RCP<Teuchos::ParameterList> validParamList;
   if(validParamList.get()==NULL) {
     validParamList = Teuchos::rcp(new Teuchos::ParameterList("BelosLinearOpWithSolveFactory"));
-    setStringToIntegralParameter<EBelosSolverType>(
-      SolverType_name, SolverType_default,
-      "Type of linear solver algorithm to use.",
-      tuple<std::string>(
-        "Block GMRES",
-        "Pseudo Block GMRES",
-        "Block CG",
-        "Pseudo Block CG",
-        "Pseudo Block Stochastic CG",
-        "GCRODR",
-        "RCG",
-        "MINRES",
-        "TFQMR"
-        ),
-      tuple<std::string>(
-        "Block GMRES solver for nonsymmetric linear systems.  It can also solve "
```

Comment:
I don't think these documentation strings exist anywhere else.  Do we want to save the strings somewhere, say in `Belos::SolverFactory` itself?

Reply:
I could move them to `Belos_Details_EBelosSolverType.cpp`..